controller/main.php

```diff
@@ -91,6 +91,14 @@ public function fetchinfo($type, $fingerprint)
 				$yaml .=  "\tProfileID: " . $data['user_id'] . "\n";
 				$yaml .=  "\tProfileName: " . $data['username'] . "\n";
 				$yaml .=  "\tProfileRank: Registered User\n";
+				$yaml .=  "\tAvatar:\n";
+				if ($avatar_data = $this->core->get_avatar_data($data))
+				{
+					$yaml .=  "\t\tSrc: {$avatar_data['src']}\n";
+					$yaml .=  "\t\tWidth: {$avatar_data['width']}\n";
+					$yaml .=  "\t\tHeight: {$avatar_data['height']}\n";
+					$yaml .=  "\t\tType: {$data['user_avatar_type']}\n";
```

Comment:
This (`avatar.driver.gravatar` or `avatar.driver.upload`) is a deep implementation detail that I doubt is needed by users. Can we please remove this to reduce the chance of future API breakage?

Reply:
This data looks output looks really forced onto the api by what phpbb uses internally. I'd just include the url here and let the consumers handle everything else.

Reply:
OK, I will remove the `Type` node.

Reply:
The `Width` and `Height` nodes however I think are useful (at least in a browser context). But if we want to simplify the API I won't be against removing them.

Reply:
I agree and would use Width and Height as part of a potential future ingame implementation. IMO we should keep those.

Reply:
Perhaps get rid of the Src node, and put the URL as the value for Avatar? Implementations could then choose to ignore the child nodes completely.

Reply:
> Perhaps get rid of the Src node, and put the URL as the value for Avatar? 

This won't translate into JSON.